src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/BlobBuilder.cs

```diff
@@ -73,6 +182,37 @@ protected virtual void FreeChunk()
             // nop
         }
 
+        private static void OnLinking(BlobBuilder left, BlobBuilder right)
+        {
+            left.OnLinking(right);
+            right.OnLinking(left);
+        }
+
+        /// <summary>
+        /// Notifies when this <see cref="BlobBuilder"/> instance is linked with another one.
+        /// </summary>
+        /// <param name="other">The other <see cref="BlobBuilder"/> instance that gets linked.</param>
+        /// <remarks>
+        /// Derived types can override this method to detect when a link is being made between two different types of
+        /// <see cref="BlobBuilder"/> and take appropriate action. It is called before the underlying buffers are
+        /// linked, for both the current and the target instance.
+        /// </remarks>
+        protected virtual void OnLinking(BlobBuilder other)
+        {
+            // nop
+        }
+
+        /// <summary>
+        /// Changes the size of the byte array underpinning the <see cref="BlobBuilder"/>.
+        /// Derived types can override this method to control the allocation strategy.
+        /// </summary>
+        /// <param name="capacity">The array's new size.</param>
+        /// <seealso cref="Capacity"/>
+        protected virtual void SetCapacity(int capacity)
+        {
+            Array.Resize(ref _buffer, Math.Max(MinChunkSize, capacity));
+        }
```

Comment:
I don't understand how subclasses are supposed to override this. They will reassign the `Buffer` property, but according to a comment in #100418, reassigning `Buffer` clears the head chunk.

What are the semantics? Should `Capacity`'s setter have additional logic?

Reply:
cc: @jaredpar